sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1407,7 +1407,15 @@ class Dataset[T] private[sql](
    */
   @scala.annotation.varargs
   def hint(name: String, parameters: Any*): Dataset[T] = withTypedPlan {
```

Comment:
Can we accept `Column` input? Then people can use whatever hint parameters they want, columns, functions, etc.

Reply:
Treating String as SQL string is too tricky. We may break things and there is no way to specify string literals as hint parameters.

Reply:
Let's refactor it then. The API needs to be explicit and future-proof. We can't just look at the existing hints today.